app/models/topic_hot_score.rb

```diff
@@ -5,6 +5,22 @@ class TopicHotScore < ActiveRecord::Base
 
   DEFAULT_BATCH_SIZE = 1000
 
+  def self.hot_topic_ids_cache
+    @hot_topic_ids_cache ||= DistributedCache.new("hot_topic_ids_cache")
+  end
+
+  def self.hottest_topic_ids
+    hot_topic_ids_cache["ids"].to_a
+  end
+
+  def self.recreate_hottest_topic_ids_cache
+    hot_topic_ids_cache.clear
+
+    hot_topic_ids_cache.defer_get_set("ids") do
```

Comment:
```suggestion
    hot_topic_ids.defer_set("ids") do
```

You can use `defer_set` here. `defer_get_set` only executes the block if an entry in the cache doesn't exists.